server/src/main/kotlin/eu/kanade/tachiyomi/source/LocalSource.kt

```diff
@@ -168,117 +185,123 @@ class LocalSource(private val context: Context) : CatalogueSource {
 //                        }
 //                    }
 //                }
-//            }
-//        }
-//
-//        return Observable.just(MangasPage(mangas.toList(), false))
-//    }
-//
-//    override fun fetchLatestUpdates(page: Int) = fetchSearchManga(page, "", LATEST_FILTERS)
-//
-//    override fun fetchMangaDetails(manga: SManga): Observable<SManga> {
-//        getBaseDirectories(context)
-//            .asSequence()
-//            .mapNotNull { File(it, manga.url).listFiles()?.toList() }
-//            .flatten()
-//            .firstOrNull { it.extension == "json" }
-//            ?.apply {
-//                val reader = this.inputStream().bufferedReader()
-//                val json = JsonParser.parseReader(reader).asJsonObject
-//
-//                manga.title = json["title"]?.asString ?: manga.title
-//                manga.author = json["author"]?.asString ?: manga.author
-//                manga.artist = json["artist"]?.asString ?: manga.artist
-//                manga.description = json["description"]?.asString ?: manga.description
-//                manga.genre = json["genre"]?.asJsonArray?.joinToString(", ") { it.asString }
-//                    ?: manga.genre
-//                manga.status = json["status"]?.asInt ?: manga.status
-//            }
-//
-//        return Observable.just(manga)
-//    }
-//
-//    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
-//        val chapters = getBaseDirectories(context)
-//            .asSequence()
-//            .mapNotNull { File(it, manga.url).listFiles()?.toList() }
-//            .flatten()
-//            .filter { it.isDirectory || isSupportedFile(it.extension) }
-//            .map { chapterFile ->
-//                SChapter.create().apply {
-//                    url = "${manga.url}/${chapterFile.name}"
-//                    name = if (chapterFile.isDirectory) {
-//                        chapterFile.name
-//                    } else {
-//                        chapterFile.nameWithoutExtension
-//                    }
-//                    date_upload = chapterFile.lastModified()
-//
+            }
+        }
+
+        return Observable.just(MangasPage(mangas.toList(), false))
+    }
+
+    override fun fetchLatestUpdates(page: Int) = fetchSearchManga(page, "", LATEST_FILTERS)
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> {
+        File(applicationDirs.localMangaRoot, manga.url).listFiles().orEmpty().toList()
+            .firstOrNull { it.extension == "json" }
+            ?.apply {
+                val reader = this.inputStream().bufferedReader()
```

Comment:
Grab the latest manga info code from Tachi to use Kotlinx.Serialization for this

Reply:
hmm, my clone is old then.